diff --git a/base/Decorate.js b/base/Decorate.js
index 3aeb851111d8335fddd245f5bb3e9654c57ce53e..dc37bb2a6cd9491637f72c57d38356afb56a3114 100644
--- a/base/Decorate.js
+++ b/base/Decorate.js
@@ -13,8 +13,8 @@ import { extractStorageArgs } from '../util/validate.js';
 import { Events } from './Events.js';
 import { findCall, findError } from './find.js';
 const PAGE_SIZE_K = 1000; // limit aligned with the 1k on the node (trie lookups are heavy)
-const PAGE_SIZE_V = 250; // limited since the data may be > 16MB (e.g. misfiring elections)
-const PAGE_SIZE_Q = 50; // queue of pending storage queries (mapped together, next tick)
+const PAGE_SIZE_V = 512; // limited since the data may be > 16MB (e.g. misfiring elections)
+const PAGE_SIZE_Q = 512; // queue of pending storage queries (mapped together, next tick)
 const l = logger('api/init');
 let instanceCounter = 0;
 function getAtQueryFn(api, { method, section }) {
